src/routing_message_filter.rs

```diff
@@ -38,15 +45,23 @@ impl RoutingMessageFilter {
 
         RoutingMessageFilter {
             incoming: MessageFilter::with_expiry_duration(incoming_duration),
+            incoming_route: MessageFilter::with_expiry_duration(incoming_duration),
```

Comment:
Alternative: add an array `[bool; MAX_ROUTES]` to `TimestampedMessage`. (`MAX_ROUTES` could be fixed, possibly even to a large number like 32. Or we could just use a `Vec`.) Set true for the indices where a message was received on that route.

Should save some memory and work?